chatsky/llm/prompt.py

```diff
@@ -40,12 +43,179 @@ class Prompt(BaseModel):
     depending on where the type of this prompt.
     """
 
+    @abstractmethod
+    async def message(self, ctx: Context) -> Message:
+        """
+        Get the message content for this prompt.
+
+        :param ctx: Current dialog context.
+        :return: Message object containing the prompt content.
+        """
+        raise NotImplementedError
+
+    async def to_langchain_messages(
+        self, 
+        ctx: Context, 
+        source: str = "human",
+        position_config: Optional[PositionConfig] = None
+    ) -> List[Union[HumanMessage, SystemMessage, AIMessage]]:
+        """
+        Convert this prompt to a list of Langchain messages.
+
+        :param ctx: Current dialog context.
+        :param source: Source of the message ["human", "ai", "system"]. Defaults to "human".
+        :param position_config: Configuration for positioning different parts of the context.
+            If None, will use default PositionConfig().
+        :return: List of Langchain message objects.
+        """
+        message = await self.message(ctx)
+        if message.text == "":
+            return []
+            
+        langchain_message = await message_to_langchain(message, ctx, source=source)
+        return [langchain_message]
```

Comment:
Why not make this method abstract and remove abstract `message` method?
This would allow to return lists of messages in subclass implementation of this method while the current base class allows only for single-message lists.